test/common/common/hex_fuzz_test.cc

```diff
@@ -0,0 +1,22 @@
+#include "common/common/hex.h"
+
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+
+namespace Envoy {
+namespace Fuzz {
+
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+  Envoy::Hex::encode(buf, len);
+  Envoy::Hex::decode(std::string(reinterpret_cast<const char*>(buf), len));
+
+  FuzzedDataProvider fuzz_data(buf, len);
+  auto decoded = Envoy::Hex::decode(Envoy::Hex::encode(buf, len));
+  FUZZ_ASSERT(fuzz_data.ConsumeBytes<uint8_t>(len) == decoded);
```

Comment:
I think calling `ConsumeBytes` with `len` will read off all `len` bytes of the input string, leaving 0 for the integer calls. (https://github.com/llvm/llvm-project/blob/c494ff05ed59712ef90ffb66bfade6bac45a0ab1/compiler-rt/include/fuzzer/FuzzedDataProvider.h#L155)

You might just want to instantiate/use the FuzzedDataProvider after this, or only read some smaller random number of bytes for the encode/decode round-trip test.

Reply:
Done. Created FuzzedDataProvider instance to void chopping.